src/mpi/coll/transports/generic/transport.c

```diff
@@ -0,0 +1,1243 @@
+        MPIR_COLL_GENERIC_vtx_t *vtx = &sched->vtcs[i];
+        vtx->state = MPIR_COLL_GENERIC_STATE_INIT;
+        /* reset the number of unfinished dependencies to be the number of incoming vertices */
+        vtx->num_unfinished_dependencies = vtx->invtcs.used;
+        /* recv_reduce tasks need to be set as not done */
+        if (vtx->kind == MPIR_COLL_GENERIC_KIND_RECV_REDUCE)
+            vtx->nbargs.recv_reduce.done = 0;
+        /* reset progress of multicast operation */
+        if(vtx->kind == MPIR_COLL_GENERIC_KIND_MULTICAST)
+            vtx->nbargs.multicast.last_complete = -1;
+    }
+    /* Reset issued vertex list to NULL.
+     * **TODO: Check if this is really required as the vertex linked list
+     * might be NULL implicitly at the end of the previous schedule use
+     * */
+    MPIR_COLL_GENERIC_reset_issued_list(sched);
+}
+
+/* Transport function to get schedule (based on the key) if it exists, else return a new schedule */
+MPIR_COLL_GENERIC_sched_t *MPIR_COLL_GENERIC_get_schedule(MPIR_COLL_GENERIC_comm_t * comm,
+                                                        void *key, int len, int tag, int *is_new)
+{
+    MPIR_COLL_GENERIC_sched_t *sched = MPIR_COLL_get_sched(comm->sched_db, key, len);
+    if (sched) {        /* schedule already exists */
+        MPL_DBG_MSG_FMT(MPIR_DBG_COLL,VERBOSE,(MPL_DBG_FDEST,"Schedule is loaded from the database[%p]\n", sched));
+        *is_new = 0;
+        /* reset the schedule for reuse */
+        MPIR_COLL_GENERIC_sched_reset(sched, tag);
+    } else {
+        *is_new = 1;
+        sched = MPIR_COLL_GENERIC_allocate_mem(sizeof(MPIR_COLL_GENERIC_sched_t));
+        MPL_DBG_MSG_FMT(MPIR_DBG_COLL,VERBOSE,(MPL_DBG_FDEST,"New schedule is created:[%p]\n", sched));
+        /* initialize the newly created schedule */
+        MPIR_COLL_GENERIC_sched_init(sched, tag);
+    }
+    return sched;
+}
+
+/* Transport function to tell that the schedule generation is now complete */
+void MPIR_COLL_GENERIC_sched_commit(MPIR_COLL_GENERIC_sched_t * sched)
+{
+}
+
+/* Transport function to tell that the schedule can now be freed
+ * and can also be removed from the database. We are not doing
+ * anything here because the schedule is stored in the database
+ * can be reused in future */
+void MPIR_COLL_GENERIC_sched_finalize(MPIR_COLL_GENERIC_sched_t * sched)
+{
+}
+
+/* Transport function to get commutativity of the operation */
+void MPIR_COLL_GENERIC_opinfo(MPIR_COLL_GENERIC_op_t op, int *is_commutative)
+{
+    MPIR_Op *op_ptr;
+
+    if (HANDLE_GET_KIND(op) == HANDLE_KIND_BUILTIN)
+        *is_commutative = 1;
+    else {
+        MPIR_Op_get_ptr(op, op_ptr);
+
+        if (op_ptr->kind == MPIR_OP_KIND__USER_NONCOMMUTE)
+            *is_commutative = 0;
+        else
+            *is_commutative = 1;
+    }
+}
+
+/* Transport function to tell if the buffer is in_place - used by some
+ * collective operations */
+int MPIR_COLL_GENERIC_isinplace(const void *buf)
```

Comment:
Since the transports are MPICH specific now, we don't need this function anymore. Just do the check inline.